abstractions/python/kiota/abstractions/authentication/base_bearer_token_authentication_provider.py

```diff
@@ -0,0 +1,42 @@
+from typing import Union
+
+from ..request_information import RequestInformation
+from .authentication_provider import AuthenticationProvider
+
+
+class BaseBearerAuthenticationProvider(AuthenticationProvider):
+    """Provides a base class implementing AuthenticationProvider for Bearer token scheme.
+
+    Args:
+        AuthenticationProvider (ABC):The abstract base class that this class implements
+    """
+    AUTHORIZATION_HEADER = "Authorization"
+
+    async def authenticate_request(self, request: RequestInformation) -> None:
+        """Authenticates the provided RequestInformation instance using the provided
+        authorization token
+
+        Args:
+            request (RequestInformation): Request information object
+        """
+        if not request:
+            raise Exception("Request cannot be null")
+        if self.AUTHORIZATION_HEADER in request.headers:
+            token = await self.get_authorization_token(request)
+            if not token:
+                raise Exception("Failed to get an authorization token")
+
+            request.headers.update({f'{self.AUTHORIZATION_HEADER}': f'Bearer {token}'})
+
+    async def get_authorization_token(self, request: RequestInformation) -> str:
```

Comment:
the design around this is being changed.  see #1013

Reply:
Thanks. Updated in https://github.com/microsoft/kiota/pull/1057/commits/86b9dd409a49e66886e6045be057778369b9a2ff